hazelcast/src/main/java/com/hazelcast/map/impl/query/PartitionScanRunner.java

```diff
@@ -99,7 +99,7 @@ public void accept(Data key, Record record) {
 
                 queryEntry.init(ss, key, value, extractors);
                 queryEntry.setRecord(record);
-                queryEntry.setMetadata(recordStore.getMetadataStore().get(key));
+                queryEntry.setMetadata(recordStore.getOrCreateMetadataStore().get(key));
```

Comment:
Multiple threads can access `getOrCreateMetadataStore` concurrently like generic and partition threads, to prevent races in these cases, here we should return `null` if there is no metadata-store created. Or simply for on-heap metadata store we can remove lazy initialization.

Reply:
Fixed avoiding lazy initialization for on-heap metadata store.